lib/modules/datasource/go/base.ts

```diff
@@ -125,7 +125,10 @@ export class BaseGoDatasource {
 
         // split the go module from the URL: host/go/module -> go/module
         const split = goModule.split('/');
-        const packageName = split[1] + '/' + split[2];
+        let packageName = split.slice(1).join('/');
```

Comment:
does it still work for monorepo modules? eg when they are in a subfolder of a repo?

Reply:
You are right. It won't work this way. My new solution relies on the go src url extracted from the inital pkg url request. By using this url the solution works for monorepos too.